ui/src/lib/yjs/useYWorkflow.ts

```diff
@@ -135,9 +135,51 @@ export default ({
       handleWorkflowIdChange,
     ],
   );
+
+  const handleWorkflowRename = useCallback(
+    (id: string, name: string) =>
+      undoTrackerActionWrapper(() => {
+        if (!name.trim()) {
+          throw new Error("Workflow name cannot be empty");
+        }
+
+        // Update local state
+        setWorkflows((w) => w.map((w) => (w.id === id ? { ...w, name } : w)));
```

Comment:
We should be updating, `data.name` instead and then in the custom node doing `data?.name || name`.

Reply:
that's for the yWorkflow nodes, you'll see the same change on 164!

Reply:
No, but here you are updating the workflow's name on the main object, not the data.name.

Reply:
As discussed this is required